splice_spark/src/main/spark2.1/com/splicemachine/spark/splicemachine/SplicemachineContext.scala

```diff
@@ -229,10 +243,69 @@ class SplicemachineContext(url: String) extends Serializable {
       execute(s"ANALYZE TABLE $tableName ESTIMATE STATISTICS SAMPLE $samplePercent PERCENT")
   }
 
+  lazy val tempDirectory = {
+    val root = options.getOrElse(SpliceJDBCOptions.JDBC_TEMP_DIRECTORY, "/tmp")
+    HConfiguration.getConfiguration.authenticationNativeCreateCredentialsDatabase()
+    val fs = FileSystem.get(HConfiguration.unwrapDelegate())
+    val path = new Path(root, getRandomName())
+    fs.mkdirs(path, FsPermission.createImmutable(Integer.parseInt("711", 8).toShort))
+
+    SpliceSpark.getSession.sparkContext.addSparkListener(new SparkListener {
+      override def onApplicationEnd(applicationEnd: SparkListenerApplicationEnd): Unit = {
+        Holder.log.info("Removing " + path)
+        fs.delete(path, true)
+      }
+    })
+    path.toString
+  }
+
+  def getRandomName(): String = {
+    val name = new Array[Byte](20)
+    new SecureRandom().nextBytes(name)
+    Bytes.toHex(name)
```

Comment:
SecureRandom().nextBytes() does not seem to guarantee uniqueness. Though the chance of name conflict is slim, theoretically it is still possible. Should we append the random name with a timestamp to make it unique?